perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -411,7 +413,7 @@ def Prepare(benchmark_spec):
   logging.info('Umount scratch disk on %s at %s', vm, mount_point)
   vm.RemoteCommand('sudo umount %s' % mount_point)
 
-  if FLAGS.device_fill_size is not '0':
```

Comment:
Percentage is still necessary because the caller does not always know the
size of the disk (PKB provisions the disk and picked the size in the
default case)

On Wednesday, October 7, 2015, Anthony Voellm aka Tony the P3rfguy <
notifications@github.com> wrote:

> In perfkitbenchmarker/benchmarks/fio_benchmark.py
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/515#discussion_r41443701>
> :
>
> > @@ -411,7 +413,7 @@ def Prepare(benchmark_spec):
> >    logging.info('Umount scratch disk on %s at %s', vm, mount_point)
> >    vm.RemoteCommand('sudo umount %s' % mount_point)
> >
> > -  if FLAGS.device_fill_size is not '0':
>
> You should also simplify this and just accept a floating point number. Not
> reason to include or accept the percent. So move from DEFINE_string to
> DEFINE_double
> … <#1504406fb5a2215f_>
> On Wed, Oct 7, 2015 at 1:35 PM, noahl <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote: In
> perfkitbenchmarker/benchmarks/fio_benchmark.py <
> https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/515#discussion_r41443204>
> : > @@ -411,7 +413,7 @@ def Prepare(benchmark_spec): > logging.info('Umount
> scratch disk on %s at %s', vm, mount_point) > vm.RemoteCommand('sudo umount
> %s' % mount_point) > > - if FLAGS.device_fill_size is not '0': I took out
> the if device_fill_size is 0 check for two reasons. First, it wasn't very
> well done: it only caught a literal "0", not "0%", or "0G", or anything
> like that. Second, it didn't gain us very much. In my testing, if you run
> fio and tell it to fill with size 0, it returns very quickly. We didn't
> gain very much, and it made the code more complex, so I think it is not
> worth it. I would be open to adding the 0 check back, but I think it should
> be a separate change. If we do that change, I'd also like to address the
> issue that if we run against a filesystem with no fill (which is now the
> default), we have an unnecessary unmount-reformat-remount step. That will
> be a bigger change, but should also produce a more significant speedup. —
> Reply to this email directly or view it on GitHub <
> https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/515/files#r41443204>
> .
> -- Anthony F. Voellm (aka Tony) Google Voice: (650) 516-7382 <
> https://www.google.com/voice/b/0?pli=1#phones> Blog:
> http://perfguy.blogspot.com/ Benchmark like a pro... PerfKitBenchmarker <
> https://github.com/GoogleCloudPlatform/PerfKitBenchmarker>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/515/files#r41443701>
> .
>